Sources/BraintreeCore/Configuration/ConfigurationLoader.swift

```diff
@@ -33,29 +35,28 @@ class ConfigurationLoader {
     /// 2. If no cached configuration is found, it fetches the configuration from the server and caches the successful response.
     /// 3. If fetching the configuration fails, it returns an error.
     ///
-    /// - Parameters:
-    ///   - completion: A completion handler that is called with the fetched or cached `BTConfiguration` object or an `Error`.
-    ///
-    /// - Completion:
-    ///   - `BTConfiguration?`: The configuration object if it is successfully fetched or retrieved from the cache.
-    ///   - `Error?`: An error object if fetching the configuration fails or if the instance is deallocated.
+    /// - Returns: A `BTConfiguration` if it is successfully fetched or retrieved from the cache.
+    /// - Throws: An `Error` describing the failure; if fetching the configuration fails or if the instance is deallocated.
     @_documentation(visibility: private)
     @ConfigurationActor
     func getConfig() async throws -> BTConfiguration {
+        if let cachedConfig = try? configurationCache.getFromCache(authorization: http.authorization.bearer) {
```

Comment:
I think we can move forward with checking the cache first, if a cache save is in flight we would use the existing task value below. Added a docstring for clarity for this.

Reply:
I was concerned about chance of data race between reading from cache and writing to cache from the Task block.
If we are doing getFromCache while there is an existingTask, we run a small risk of simultaneous read/write to cache.

Reply:
There should not be a data race in that case, in that case the sequence of events would be:
1. We check the cache and nothing is there because we are actively writing to the cache
2. There _is_ and `existingTask` so we would return that on like 49

There wouldn't be an instance were we are reading and writing at the same time because `getFromCache` would be empty in this case while `existingTask` would not

Reply:
I was thinking about the rare case where we are reading from cache and there is an existing task that is writing into cache while it's reading cache. Simultaneous read/write cache would result.

Reply:
`NSCache` is multi-threading friendly so that shouldn't be an issue. But also, if we were writing to the cache at the exact same moment as attempting to read from the cache the object would not yet exist in the `NSCache` so we would instead return the `existingTask.value`. The `NSCache` also only stores a singular object per key, which is the `authorizationFingerprint` or `tokenizationKey`.

Reply:
I didn't know that about NSCache! Thanks.